packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1317,19 +1319,23 @@ export class ContainerRuntime extends EventEmitter
 
     public async _createDataStoreWithProps(pkg: string | string[], props?: any, id = uuid()):
         Promise<IFluidDataStoreChannel> {
-        return this._createFluidDataStoreContext(Array.isArray(pkg) ? pkg : [pkg], id, props).realize();
+        return this._createFluidDataStoreContext(Array.isArray(pkg) ? pkg : [pkg], id, false, props).realize();
     }
 
-    private async _createDataStore(pkg: string | string[], id = uuid()): Promise<IFluidDataStoreChannel>
+    private async _createDataStore(
+        pkg: string | string[],
+        id = uuid(),
+        root: boolean = false,
+        ): Promise<IFluidDataStoreChannel>
     {
-        return this._createFluidDataStoreContext(Array.isArray(pkg) ? pkg : [pkg], id).realize();
+        return this._createFluidDataStoreContext(Array.isArray(pkg) ? pkg : [pkg], id, root).realize();
     }
 
     private canSendOps() {
         return this.connected && !this.deltaManager.readonly;
     }
 
-    private _createFluidDataStoreContext(pkg: string[], id: string, props?: any) {
+    private _createFluidDataStoreContext(pkg: string[], id: string, root: boolean = false, props?: any) {
```

Comment:
Consider not making root a default argument - that's rather important part of contract that callers should be very explicit about.